contracts/src/lunarswap-v1/LunarswapLiquidity.compact

```diff
@@ -0,0 +1,238 @@
+ * entitle holders to a portion of trading fees and the underlying assets.
+ * 
+ * Key Features:
+ * - LP token minting for liquidity providers
+ * - LP token burning for liquidity removal
+ * - Total supply tracking per trading pair
+ * - Token metadata management (name, symbol, decimals)
+ * - Nonce-based token generation
+ * - UTXO-based token model
+ */
+module LunarswapLiquidity {
+  import CompactStandardLibrary;
+
+  import "../../../compact-contracts/contracts/src/archive/ShieldedToken" prefix ShieldedToken_;
+  import "../math/Uint128" prefix Uint128_;
+  import "../math/Bytes32" prefix Bytes32_;
+
+  /**
+   * @ledger _counter
+   * @description
+   * Internal counter used for LP token nonce evolution and minting operations.
+   */
+  ledger _counter: Counter;
+
+  /**
+   * @ledger lpTokenNonce
+   * @description
+   * The global nonce used for LP token generation and uniqueness.
+   */
+  export ledger lpTokenNonce: Bytes<32>;
+
+  /**
+   * @ledger lpTotalSupply
+   * @description
+   * Mapping of trading pair hash (pairId) to the total supply and metadata of LP tokens for that pair.
+   * The key is the pair hash (Bytes<32>), and the value is the QualifiedCoinInfo for the LP token.
+   */
+  export ledger lpTotalSupply: Map<Bytes<32>, QualifiedCoinInfo>;
+
+  /**
+   * @ledger lpTokenName
+   * @description
+   * The name of the LP token (e.g., "Lunarswap LP Token").
+   */
+  export ledger lpTokenName: Opaque<"string">;
+
+  /**
+   * @ledger lpTokenSymbol
+   * @description
+   * The symbol of the LP token (e.g., "LUNAR-LP").
+   */
+  export ledger lpTokenSymbol: Opaque<"string">;
+
+  /**
+   * @ledger lpTokenDecimals
+   * @description
+   * The number of decimals used for LP tokens (e.g., 18).
+   */
+  export ledger lpTokenDecimals: Uint<8>;
+
+  /**
+   * @ledger lpTokenType
+   * @description
+   * The color/type identifier for the LP token (Bytes<32>).
+   */
+  export ledger lpTokenType: Bytes<32>;
+
+  /**
+   * @title initialize circuit
+   * @description Initializes the LP token system with basic configuration.
+   * 
+   * @remarks
+   * This circuit sets up the initial configuration for LP tokens including name,
+   * symbol, decimals, and nonce. It should only be called once during contract deployment.
+   *
+   * @param {Bytes<32>} nonce_      - The nonce for LP token generation.
+   * @param {Opaque<"string">} name_ - The name of the LP token.
+   * @param {Opaque<"string">} symbol_ - The symbol of the LP token.
+   * @param {Uint<8>} decimals_     - The number of decimals for LP tokens.
+   *
+   * @returns [] - No return values.
+   */
+  export circuit initialize(
+                   nonce_: Bytes<32>,
+                   name_: Opaque<"string">,
+                   symbol_: Opaque<"string">,
+                   decimals_: Uint<8>
+                   ): [] {
+    lpTokenNonce = nonce_;
+    lpTokenName = name_;
+    lpTokenSymbol = symbol_;
+    lpTokenDecimals = decimals_;
+    return [];
```

Comment:
We can use `initializable` pattern here from the contracts lib, no?

Reply:
Though, if you agree with making the metadata sealed, I wonder if it makes sense to prevent re-initialization because the compiler won't allow contracts to export circuits that modify `sealed` vars